tedana/io.py

```diff
@@ -501,6 +504,195 @@ def filewrite(data, filename, ref_img, gzip=False, copy_header=True):
+        is components and `T` is the same as in `data`
+    n_vols : :obj:`int`
+        Number of volumes in original time series
+    acc : :obj:`list`
+        Indices of accepted (BOLD) components in `mmix`
+    rej : :obj:`list`
+        Indices of rejected (non-BOLD) components in `mmix`
+    midk : :obj:`list`
+        Indices of mid-K (questionable) components in `mmix`
+    empty : :obj:`list`
+        Indices of ignored components in `mmix`
+    ref_img : :obj:`str` or img_like
+        Reference image to dictate how outputs are saved to disk
+
+    """
+
+    # regenerate the beta images
+    ts_B = model.get_coeffs(ts, mmix, mask)
+    ts_B = ts_B.reshape(ref_img.shape[:3] + ts_B.shape[1:])
+    # Mask out zeros
+    ts_B = np.ma.masked_where(ts_B == 0, ts_B)
+
+    # Get repitition time from ref_img
+    tr = ref_img.header.get_zooms()[-1]
+
+    # Start making some really ugly pluts
+    import os
+    if not os.path.exists('./figures'):
+        os.mkdir('figures')
+
+    os.chdir('./figures')
+
+    # This precalculates the Hz for the fft plots
+    Fs = 1.0/tr
+    # resampled frequency vector
+    f = Fs * np.arange(0, n_vols // 2 + 1) / n_vols
+
+    # Create indices for 6 cuts, based on dimensions
+    xdim = ts_B.shape[0]
+    xcut = int(xdim/6)
+
+    ydim = ts_B.shape[1]
+    ycut = int(ydim/6)
+
+    zdim = ts_B.shape[2]
+    zcut = int(zdim/6)
+
+    for compnum in range(0, mmix.shape[1], 1):
+
+        allplot = plt.figure(figsize=(10, 9))
+        ax_ts = plt.subplot2grid((5, 6), (0, 0), rowspan=1, colspan=6,
+                                 fig=allplot)
+        if compnum in acc:
+            line_color = 'g'
+        elif compnum in rej:
+            line_color = 'r'
+        elif compnum in midk:
+            line_color = 'm'
+        else:
+            line_color = 'k'
+
+        ax_ts.plot(mmix[:, compnum], color=line_color)
+
+        # Title will include variance from comptable
+        comp_var = "{0:.2f}".format(comptable.iloc[compnum][3])
+        plt_title = 'Comp. ' + str(compnum) + ': ' + comp_var + '% variance'
+        title = ax_ts.set_title(plt_title)
+        title.set_y(1.5)
+        ax_ts.set_xlabel('TRs')
+        ax_ts.set_xbound(0, n_vols)
+        # Make a second axis with units of time (s)
+        max_xticks = 10
+        xloc = plt.MaxNLocator(max_xticks)
+        ax_ts.xaxis.set_major_locator(xloc)
+
+        ax_ts2 = ax_ts.twiny()
+        ax1Xs = ax_ts.get_xticks()
+
+        ax2Xs = []
+        for X in ax1Xs:
+            ax2Xs.append(X * tr)
+        ax_ts2.set_xticks(ax1Xs)
+        ax_ts2.set_xbound(ax_ts.get_xbound())
+        ax_ts2.set_xticklabels(ax2Xs)
+        ax_ts2.set_xlabel('seconds')
+
+        # Set range to ~1/10th of max beta
+        imgmax = ts_B[:, :, :, compnum].max()*.1
+        imgmin = ts_B[:, :, :, compnum].min()*.1
+
+        count = 0
+        for imgslice in range(xcut, xdim-xcut, xcut):
```

Comment:
Can we have space around the operator, here ? Also, it looks like this can just be a function that we iterate through `cuts` for, rather than repeating three times !

Reply:
I simplified this, but perhaps not as clean as you had in mind. Now a single loop.